src/renderer/entities/transactionBuilder/lib/helpers/helpers.ts

```diff
@@ -0,0 +1,88 @@
+import {AccountsInWallet, asMany} from "@shared/core/types/wallet";
+import {
+  CompoundWalletVisit,
+  LeafVisit,
+  MultisigVisit,
+  TransactionBuilder
+} from "@entities/transactionBuilder/model/transaction-builder";
+import {AmountReduction, buildAmountReduction} from "@entities/transactionBuilder/lib/helpers/amount-reduction";
+import {ApiPromise} from "@polkadot/api";
+import {UnsignedTransaction} from "@substrate/txwrapper-polkadot";
+import {createTxMetadatas, toAddress} from "@shared/lib/utils";
+
+export function getSigningAccounts(transactionBuilder: TransactionBuilder): AccountsInWallet | undefined {
+  let signingAccounts: AccountsInWallet | undefined = undefined
+
+  transactionBuilder.visitSelf({
+    visitLeaf(visit: LeafVisit) {
+      signingAccounts = asMany(visit.account)
+    },
+
+    visitMultisig(visit: MultisigVisit) {
+      signingAccounts = asMany(visit.selectedSignatory)
+    },
+
+    visitCompoundWallet(visit: CompoundWalletVisit) {
+      signingAccounts = {
+        wallet: visit.wallet,
+        accounts: visit.selectedChildrenAccounts
+      }
+    }
+  })
+
+  return signingAccounts
+}
+
+export async function getTransactionFee(transactionBuilder: TransactionBuilder): Promise<AmountReduction | undefined> {
+  const signingAccounts = getSigningAccounts(transactionBuilder)
+  if (signingAccounts == undefined) return undefined
+
+  const singleTx = await transactionBuilder.submittableExtrinsic()
+  const singleTxFee = await singleTx?.paymentInfo(signingAccounts.accounts[0].accountId)
+  if (singleTxFee == undefined) return undefined
+
+  const reductionBuilder = buildAmountReduction()
+
+  signingAccounts.accounts.forEach((account) => {
+    reductionBuilder.addReductionAmount(account.accountId, singleTxFee.partialFee.toBigInt())
+  })
+
+  return reductionBuilder.build()
+}
+
+export async function getDeposits(transactionBuilder: TransactionBuilder): Promise<AmountReduction> {
+  const reductionBuilder = buildAmountReduction()
+
+  const multisigDeposit = await getMultisigDeposit(transactionBuilder.api)
+
+  transactionBuilder.visitAll({
+    visitMultisig(visit: MultisigVisit) {
+      // TODO check who pays deposit in case multisig is wrapped in some other structure, like proxy
```

Comment:
If that helps, the submitter always pays the deposit for the multisig transaction state. Now depending on the action performed, there may be other deposit required. for instance:
- A multisig is creating a proxy. The submitter needs ~20 DOTs for the multisig tx, but the Multisig that will then create the proxy will also need 20 DOTs for the proxy creation deposit
- A Multisig with pure proxy sets an identity. The submitter needs ~20 DOTs for the multisig tx, but the Multisig that will then set an will also need 20 DOTs for the identity deposit

TL;DR, the depositor/submitter needs the multisig tx deposit, but the transaction may fail when being submitted by the last signatory, because of some other deposit.